File: Server/src/Models/Civilian.cs
using GTANetworkServer;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GTAIdentity.Models
{
    [Serializable]
    public class Civilian : Character
    {
        public Wanted WantedLevel { get; set; }
        public decimal Cash { get; set; }
        public Job Job { get; set; }

        public Civilian(string name,Client client,Account account) : base(name,client,account) { }
        public Civilian(string name,Client client,Guid account) : base(name,client,account) { }

        public override string GetStats()
        {
            var info = base.GetStats() + "$\n" +
                $"Wanted level: {WantedLevel}, Cash: {Cash}";
            return info;
        }

        public override string ToString()
        {
            return $"Civilian {Name}";
        }

        protected Civilian() : base()
        {
        }
    }
    [Serializable]
    public enum Job
    {
        Unemployed = 0,
        Thief = 1,

    }
    [Serializable]
    public enum Wanted
    {
        None = 0,
        Ticketable = 1,
        Arrestable = 2,
        KillAuthorized = 3
    }
}

File: Server/types-gtaidentity/Civilian.ts


/// <reference path="../types-gtanetwork/index.d.ts" />

    // $Classes/Enums/Interfaces(filter)[template][separator]
    // filter (optional): Matches the name or full name of the current item. * = match any, wrap in [] to match attributes or prefix with : to match interfaces or base classes.
    // template: The template to repeat for each matched item
    // separator (optional): A separator template that is placed between all templates e.g. $Properties[public $name: $Type][, ]

    // More info: http://frhagn.github.io/Typewriter/
    

export interface Civilian {
    
    wantedLevel: Wanted;
    cash: number;
    job: Job;
}


export enum Job {
    
    unemployed = 0,
    thief = 1
}
export enum Wanted {
    
    none = 0,
    ticketable = 1,
    arrestable = 2,
    killAuthorized = 3
}
	


